shopping_mall/backend/ai/agent/cs_tools.py

```
@@ -156,40 +180,86 @@ class RefuseRequestInput(BaseModel):
-            docs = rag_service.retrieve(nq, "storage_guide", top_k=3, distance_threshold=settings.rag_storage_retry_threshold)
-        return "\n\n".join(docs) if docs else f"'{product_name}' 보관법 정보를 찾을 수 없습니다."
 
-    async def search_season_info(query: str, season: str | None = None) -> str:
-        nq = normalize_query(query)
-        where = {"season": season} if season else None
-        docs = rag_service.retrieve(nq, "season_info", top_k=3, distance_threshold=settings.rag_distance_threshold, where=where)
-        return "\n\n".join(docs) if docs else "제철 정보를 찾을 수 없습니다."
+        # 서브카테고리 필터로 결과 없으면 전체 재시도
+        if not pairs and subcategory:
+            logger.info(
+                f"[search_faq] subcategory='{subcategory}' 결과 없음 → 전체 FAQ 재검색 (fallback)"
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)